src/models/generated/GroupReserve.jl

```diff
@@ -0,0 +1,84 @@
+#=
+This file was not auto-generated.
```

Comment:
This code should get auto-generated if you make appropriate changes to [power_system_structs.json](https://github.com/NREL-SIIP/PowerSystems.jl/blob/master/src/descriptors/power_system_structs.json) and execute the following line of code:

```julia
InfrastructureSystems.generate_structs("./src/descriptors/power_system_structs.json", "./src/models/generated")
```

Reply:
You don't need to generate this function. The methods should work automatically.

Reply:
uhmm but the proposed `get_contributing_device_mapping` calls `_get_contributing_devices` which for normal service does the following:
```
function _get_contributing_devices(sys::System, service::T) where {T <: Service}
    uuid = IS.get_uuid(service)
    devices = ServiceContributingDevices(service, Vector{Device}())
    for device in get_components(Device, sys)
        for _service in get_services(device)
            if IS.get_uuid(_service) == uuid
                push!(devices.contributing_devices, device)
                break
            end
        end
    end
    return devices
end
```
but since the contributing services are inside my struct I have to treat in differently.

Reply:
should I move this function to base.jl?

Reply:
done

Reply:
Yes, in the same place the `current get_contributing_devices`